### ISSUE TITLE
Could I set some initial guess for camera intrinsic and extrinsic parameters?

### ISSUE THREAD

**Issue by fred2019-dev** [colmap/colmap#664]
I've got some coarse estimations of the extrinsic and intrinsic parameters of several views from ARCore, and images taken at those views. 

Could I let colmap extract features of the corresponding images, match them, then use the matched features, together with those coarse estimations as initial guess, to get better estimations?

Thanks very much!

**Comment by tsattler** [comment]
Yes, you can. Either in the GUI when extracting features, on the command line when running feature_extractor, or by modifying the intrinsics stored in the database file.

**Comment by fred2019-dev** [comment]
> Yes, you can. Either in the GUI when extracting features, on the command line when running feature_extractor, or by modifying the intrinsics stored in the database file.

Thanks very much!
Could I also modify the extrinsics stored in database file, and let them also be part of the initial guess for the mapper?

**Comment by tsattler** [comment]
@fred2019-dev The mapper does not use any extrinsics stored in the database for initialization. If you have accurate poses, you can create a 3D reconstruction by hand that only contains the camera extrinsics and intrinsics (but no 3D points) and use the feature_triangulator to recover the 3D structure. You can then continue the mapping process with the resulting 3D model (see also #658 ).

**Comment by fred2019-dev** [comment]
> @fred2019-dev The mapper does not use any extrinsics stored in the database for initialization. If you have accurate poses, you can create a 3D reconstruction by hand that only contains the camera extrinsics and intrinsics (but no 3D points) and use the feature_triangulator to recover the 3D structure. You can then continue the mapping process with the resulting 3D model (see also #658 ).

@tsattler 
So I first use colmap feature extractor and colmap matcher to retrieve the matched features in all views, and then run colmap feature_triangulator to get point cloud from the matched features. After that, I could just run the colmap mapper to further refine the parameters?

Thanks very much!

**Comment by tsattler** [comment]
@fred2019-dev Once you have extracted features and computed matches, you can use the point_triangulator to compute the 3D structure from known poses. For this, you will need to create a Colmap reconstruction in which you store the poses of the images that you know.

**Comment by fred2019-dev** [comment]
> @fred2019-dev Once you have extracted features and computed matches, you can use the point_triangulator to compute the 3D structure from known poses. For this, you will need to create a Colmap reconstruction in which you store the poses of the images that you know.

@tsattler 
Thanks very much, I followed your guide and did get good result from point_triangular. But if then I would like to refine the camera parameters, what should I do? Should I use bundle_adjuster or mapper? I tried mapper and noticed that a bunch of .bin files (for point_triangulator i used .txt files) were generated, while their values are quite different from the initial values. 
Maybe the mapper didn't take the .txt files I placed in the output path as input at all?

**Comment by tsattler** [comment]
@fred2019-dev bundle_adjuster will just run a bundle adjustment on the reconstruction while mapper will try to continue the reconstruction from the model. Both normalize the reconstruction before bundle adjustment, resulting in a change in scale. See also #595 .

**Comment by rutvikokate** [comment]
> I've got some coarse estimations of the extrinsic and intrinsic parameters of several views from ARCore, and images taken at those views.
> 
> Could I let colmap extract features of the corresponding images, match them, then use the matched features, together with those coarse estimations as initial guess, to get better estimations?
> 
> Thanks very much!

How did you extract extrinsic parameters with arcore? Is there an already developed app for that? Thanks

**Comment by ahojnnes** [comment]
Closing as the question about setting initial camera parameters was thoroughly answered, explaining feature_extractor flags, database modification, and point_triangulator for known poses.
